### ISSUE TITLE
GameGuru MAX - Request - Means of Installing Exp. Build Separately from Public Release

### ISSUE THREAD

**Issue by MonkeyFrogStudio** [TheGameCreators/GameGuruRepo#5063]
I have no idea how this could be done or even if it could be done, but I think it would be great to be able to have both the latest public build and the latest experimental build installed at the same time. Why? Glad you asked! 

Let's say I'm working on a serious project in MAX. I'll want the latest public build so I can ensure nothing changes/gets broken while in development. But that would mean I could no longer test any experimental features. If my project takes a year or two (or more) to complete, then I would be out of the loop when it comes to testing MAX experimental builds, reporting bugs, making requests, etc.

If we could have a separate experimental build install, then we could keep developing while also testing without fear of the experimental build breaking our games. 

Currently, it's not much of an issue because, frankly, MAX is not really where it needs to be to create a serious project (imo). But this may become an issue later on. For example, let's say I am developing a game in MAX v1 (some future, final public release) but MAX v2 is in testing via the experimental builds. I would most likely want to keep my game development locked to v1, but I would still want to test v2 experimental. 

Again, not sure this is possible with Steam, but it would be nice. Every software I've beta tested for has the means to install any "experimental" versions separately from the main or full release.

**Comment by synchromesh62** [comment]
I actually tried this and they just conflict .. loads of false errors and load a map ny one and it shows the last loaded map in the other .. In fact Classic does the same thing.

**Comment by Necrym59** [comment]
Versioning is extremly a bad idea - even with a project system in place to structure it you then have the problems of one version having to be incompatible with the other version to keep stability, so it doesnt help. Software that is self contained to doing one taske such as creating a creating a graphic or model entity, can have various versions if the end product formats dont change, but a software that is is itself creating a end software product is another matter. Even unity, unreal have versions that are inherently incompatible with previous versions so it makes the whole point moot. TGC would have to release a whole new arm of the product on steam and it would be counter productive overall, especially since one version is an incremental update from the other.

**Comment by MonkeyFrogStudio** [comment]
Like I said, I didn't know if it were even possible, especially given that Steam is the medium by which these builds are delivered. Having said that, even graphics programs, especially 3D modeling software, often releases newer builds that are incompatible with older builds (i.e. the old format won't or may not load in the newer version and the newer version formats definitely will not load in the older format). And by being able to install them in different locations on your computer, you can keep the two completely separate so you can still be productive with the last public release while still testing the latest alpha/beta/RC.

**Comment by MonkeyFrogStudio** [comment]
Just looked into it and it turns out that Unreal, Unity, Flax, etc. all allow the end-user to install different versions of their software simultaneously. Thus, if you're working on an Unreal 4 project, you can also have Unreal 5 installed. And if you have Unreal 4.X installed, you can also have Unreal 4.Y installed, etc. When you launch a project, it launches the engine it's associated with. Thus, it's perfectly fine to have several versions of any of these 'development products' installed at the same time. 

Here's a screen shot of my Flax installation showing that I have two versions installed with the window open allowing me to select another to install if I want to:

![multi'](https://github.com/TheGameCreators/GameGuruRepo/assets/36735507/3d02bbce-1628-4916-85aa-783344b2b3af)

Back when I used to use 3DGameStudio (long time ago), I was a beta tester for them. As a result, I had both the 'final' release version installed and the beta version. That way I could still work on my projects with the 'final' release and also test the software via the beta installation. 

This is pretty standard across all forms of software, actually.

The issue, in this case, is Steam.

**Comment by GraPhiX-Guru** [comment]
It is possible to have both versions installed, not the way you may want it though, i use VM's for debugging whilst i can get Classic to work in a VM MAX does not like it, so i have a Development PC and a 'Home' PC the development pc runs beta and my home runs retail, this gives me the added bonus of trying different hardware platforms with software, this is not ideal having 2 computers for some people but it works quite well for me, and i think most households have more than one PC these days.

**Comment by Necrym59** [comment]
Yes what your asking may happen after a project system is in place but frankly it isnt worth having two experimental versions around at present until one is stable enough to be a true retail edition.

**Comment by MonkeyFrogStudio** [comment]
I don't want to install two experimental versions. Never said that. I (eventually) want to install a Public Release (non-experimental) and an experimental. That way, once I start to work on my game, I can use ONE version (the 'true retail edition') but also continue to test the latest experimental version.

Even now we have various releases we can choose from, one of which is supposed to be a public release (i.e. non-experimental version). 

So, you're correct that it's not worth it NOW, but it will be worth it later. It's one of the reasons, in my original post at the top, that I mentioned v1 and v2 of MAX as an example.

**Comment by MonkeyFrogStudio** [comment]
It's not that important? I completely disagree. As pointed out, when someone is working on a game in MAX, they are going to want to stay with the last solid or final release of the product. They are not going to want to use an experimental/alpha/beta version for development. This being the case, they would be taken out of beta testing while in development. And that's a shame. There's no reason a developer, like me, shouldn't be able to do both (work on their project with a solid release AND beta test an experimental version). There's no reason to take someone out of the beta loop simply because they are developing their game.

**Comment by MonkeyFrogStudio** [comment]
@GraPhiX-Guru That is a solution. Unfortunately, I don't have two PCs. But it's an interesting idea. Thanks. :)

**Comment by MonkeyFrogStudio** [comment]
@synchromesh62 Correct. This is why I said above, "The issue, in this case, seams to be Steam" (my second response at the very bottom).

**Comment by synchromesh62** [comment]
Its also worth noting my first response .. I tried and they conflicted.
I installed the retail .. Copied it to another drive .. Then installed the EXP, They do not like each other.
settings get saved in the second copy Last map loaded is there despite not even having the map in the other version.
Im not saying it could not be done if Lee rectified those problems but is it really worth it with so much more needed right now.

**Comment by MonkeyFrogStudio** [comment]
@synchromesh62 Yes, you are correct. There is a ton more that's needed right now. But keep in mind, this is a REQUEST. Like all the other requests made on GitHub, it's not an emergency and does not have to be addressed immediately or even in the near future. I was not demanding it be done ASAP. ;) I was just asking if it could be done and stating why it would be of benefit. Just because other things have a much higher priority than this request does not mean other requests cannot be made. After all, there are requests on this GitHub that are a few years old already ... ;)
